test/HcalDigiPipelineTest.cxx

```diff
@@ -0,0 +1,259 @@
+    ntuple_.addVar<float>("HcalDigiTest","RecY");
+    ntuple_.addVar<float>("HcalDigiTest","RecZ");
+    
+    ntuple_.addVar<int>("HcalDigiTest", "DaqDigi");
+    ntuple_.addVar<int>("HcalDigiTest", "DaqDigiIsADC");
+    ntuple_.addVar<int>("HcalDigiTest", "DaqDigiADC");
+    ntuple_.addVar<int>("HcalDigiTest", "DaqDigiTOT");
+  }
+
+  void analyze(const framework::Event &event) final override {
+
+    const auto simHits = event.getCollection<ldmx::SimCalorimeterHit>("HcalSimHits");
+
+    REQUIRE(simHits.size() == 1);
+
+    float truth_energy = simHits.at(0).getEdep();
+    ntuple_.setVar<float>("SimEnergy", truth_energy);
+    ntuple_.setVar<float>("SimX", simHits.at(0).getPosition()[0]);
+    ntuple_.setVar<float>("SimY", simHits.at(0).getPosition()[1]);
+    ntuple_.setVar<float>("SimZ", simHits.at(0).getPosition()[2]);
+    
+    const auto daqDigis{
+      event.getObject<ldmx::HgcrocDigiCollection>("HcalDigis")};
+    auto daqDigi = daqDigis.getDigi(0);
+    ntuple_.setVar<int>("DaqDigi",daqDigi.soi().raw());
+    bool is_in_adc_mode = daqDigi.isADC();
+    ntuple_.setVar<int>("DaqDigiIsADC",is_in_adc_mode);
+    ntuple_.setVar<int>("DaqDigiADC",daqDigi.soi().adc_t());
+    ntuple_.setVar<int>("DaqDigiTOT",daqDigi.tot());
+
+    const auto recHits = event.getCollection<ldmx::HcalHit>("HcalRecHits");
+    CHECK(recHits.size() == 1);
+
+    auto hit = recHits.at(0);
+    ldmx::HcalID id(hit.getID());
+    CHECK_FALSE(hit.isNoise());
+    CHECK(id.raw() == simHits.at(0).getID());
+
+    ntuple_.setVar<float>("RecX", hit.getXPos());
+    ntuple_.setVar<float>("RecY", hit.getYPos());
+    ntuple_.setVar<float>("RecZ", hit.getZPos());
+    
+    // define target energy by using the settings at the top
+    auto target_daq_energy = Approx(truth_energy).epsilon(MAX_ENERGY_PERCENT_ERROR_DAQ_TOT_MODE/100);
+    if (is_in_adc_mode) target_daq_energy = Approx(truth_energy).margin(MAX_ENERGY_ERROR_DAQ_ADC_MODE);
+
+    std::cout << "get ampl " << hit.getEnergy() << " taget daq " << truth_energy << " + " << MAX_ENERGY_ERROR_DAQ_ADC_MODE << std::endl;
+    CHECK(hit.getEnergy() == target_daq_energy);
+    ntuple_.setVar<float>("RecEnergy", hit.getEnergy());
+    
+    std::cout << "get x " << hit.getXPos() << " y " << hit.getYPos() << " z " << hit.getZPos() << std::endl;
+    
+    return;
+  }
+}; // HcalCheckEnergyReconstruction
+
+} // namespace test
+} // namespace hcal
+
+DECLARE_PRODUCER_NS(hcal::test, HcalFakeSimHits)
+DECLARE_ANALYZER_NS(hcal::test, HcalCheckEnergyReconstruction)
+
+/**
+ * Test for the Hcal Digi Pipeline
+ *
+ * Does not check for realism. Simply makes sure sim energies
+ * end up being "close" to output rec energies.
+ *
+ * Checks
+ *  - Keep reconstructed energy depositied close to simulated value
+ *  - Keep estimated energy at TP level close to simulated value
+ *
+ * @TODO still need to expand to multiple contribs in a single sim hit
+ * @TODO check layer weights are being calculated correctly somehow
+ */
```

Comment:
Update this documentation for what this hcal test _actually_ checks and what the hcal folks (you!) still have todo.